### PR TITLE
Refactor MemoryUsed to CurDirtyBytes

### PR DESCRIPTION
**Pull request by abhinavdangeti**
This API is designed to only emit the dirty "unpersisted"
bytes only. This does not included mmap'ed zap segments.

### CODE REVIEW

File: index/scorch/scorch.go [hunk 1]
```diff
@@ -407,16 +407,20 @@ func (s *Scorch) AddEligibleForRemoval(epoch uint64) {
...
 	s.rootLock.RLock()
-	for _, segmentSnapshot := range s.root.segment {
-		memUsed += 8 /* size of id -> uint64 */ +
-			segmentSnapshot.segment.SizeInBytes()
-		if segmentSnapshot.deleted != nil {
-			memUsed += segmentSnapshot.deleted.GetSizeInBytes()
+	if s.root != nil {
+		for _, segmentSnapshot := range s.root.segment {
+			memUsed += 8 /* size of id -> uint64 */
+			if _, ok := segmentSnapshot.segment.(*mem.Segment); ok {
```
Comment (sreekanth-cb) [inline index/scorch/scorch.go]: sorry, I am not able to follow the logic here,  why are we restricting the memory used computations only to mem segments? At any point in time , there could be a mix of both mem and zap segments within an index snapshot right? 
I could see in the earlier logic, we were calling the segment.SizeInBytes() API which was implemented by both mem and zap segments..

Comment (sreekanth-cb) [reply index/scorch/scorch.go]: Whoops...Nevermind, I read the commit msg now.. :)

**Comment by mschoch** [comment]
So, I'm still not 100% sold on ignoring the heap space used by the disk segments (not mmap'd space).  It seems to me like the sizes we're talking about here are tiny.  Like 100 bytes + the size of the field data.  The tests we're doing are really small scale, so even with 512MB, we should be able to have LOTS of disk segments before this is a concern.

Can't we just do a simpler change to stop include 'len(s.mm)' in the calculation, as that was not something I intended us to include and missed in the review phase.

**Comment by abhinavdangeti** [comment]
Yeah that makes sense, I'll push up another patch - also to the right branch.

**Pull request closed without merging**